### DISCUSSION TITLE
Mkdocs Build Unit Test

### DISCUSSION THREAD

**Discussion by GBBBAS** [mkdocs/mkdocs#3288] [General]
Does anyone know how to perform `mkdocs build --strict` inside a unit test?

I have tried to replicate the code as per the click command https://github.com/mkdocs/mkdocs/blob/master/mkdocs/__main__.py#L278 
```
    """Build the MkDocs documentation"""
    from mkdocs.commands import build

    _enable_warnings()
    cfg = config.load_config(**kwargs)
    cfg.plugins.on_startup(command='build', dirty=not clean)
    try:
        build.build(cfg, dirty=not clean)
    finally:
        cfg.plugins.on_shutdown()
```

However, it fails as `on_startup` is not an available function for `cfg.plugins`. I am guessing there is a kwarg that ensures `cfg` is instantiated properly, but I've not managed to work it out. Any ideas or suggestions on how to perform this build from code instead of the cli?

**Comment by pawamoy** [comment]
Here's what we have in mkdocstrings:

```python
mkdocs_config = load_config()
mkdocs_config["plugins"].run_event("startup", command="build", dirty=False)
try:
    build(mkdocs_config)
finally:
    mkdocs_config["plugins"].run_event("shutdown")
```

**Reply by GBBBAS** [reply]
perfect, that works! thanks!

I also added `strict=True` to `load_config()` to get the same effect as `--strict`

**Comment by oprypin** [comment]
This is not a "unit" test anyway. Please just run the command `mkdocs build --strict`
